libs/stats/odc/stats/tasks.py

```diff
@@ -459,9 +515,29 @@ def stream_from_sqs(
             sqs_queue = get_queue(sqs_queue)
 
         for msg in get_messages(sqs_queue, visibility_timeout=visibility_timeout, **kw):
-            # TODO: switch to JSON for SQS message body
             token = SQSWorkToken(msg, visibility_timeout)
-            tidx = parse_task(msg.body)
+            tidx, filedb = parse_sqs(msg.body)
+
+            # avoid the download and update again, how ever, we have to setup an exception to handle the unit test filedb
+            if urlparse(filedb).scheme == 's3':
+                bucket, key = s3_url_parse(filedb)
+            else: # if it is the test_tiles.db, we load it from local
+                key = filedb 
+            local_cache_file = key.split("/")[-1]
+            if not os.path.isfile(local_cache_file):  # use the download filedb from S3 as the init context flag
+                self.init_from_sqs(filedb)
+                # first time to access the filedb, then it can do the resolution check
+                if self.resolution is not None:
+                    _log.info(f"Changing resolution to {resolution[0], resolution[1]}")
```

Comment:
Not sure if I'm missing something, but it looks like `resolution` and `cfg` aren't defined here.

Reply:
You are right, I should pass these value here: https://github.com/opendatacube/odc-tools/blob/feature/move-db-file-sqs/libs/stats/odc/stats/proc.py#L67

Reply:
the _cfg_ is here: https://github.com/opendatacube/odc-tools/blob/feature/move-db-file-sqs/libs/stats/odc/stats/tasks.py#L439